src/Response.php

```diff
@@ -26,7 +26,7 @@ public function buildResponse() {
 			. 'xml';
 
 		if(!file_exists($path)) {
-			header('HTTP/1.1 404 Not Found');
+			header("Etag: " . md5(''));
 			return;
```

Comment:
Should we not return an empty JSON array? As this is what is expected by the code I think.

Reply:
XML. We first check whether the response payload is empty or not, even on a 200. https://github.com/nextcloud/server/blob/master/lib/private/Updater/ChangesCheck.php#L135 Putting there something else seems more risky actually.

Reply:
It behaves, neither admins nor users get to see something weird

Reply:
although, the user page would actually show stuff, just it cannot because that JS script errors. But there is no way around, this needs server fixes to  be clean.

Reply:
then lets just do that in server indeed :)

Reply:
https://github.com/nextcloud/server/pull/19465

Reply:
not critical, to notice you need to check the browser console.